fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/SubqueryTransformer.java

```diff
@@ -104,6 +104,43 @@ public ScalarOperator rewriteSubqueryScalarOperator(Expr predicate, OptExprBuild
         return scalarPredicate;
     }
 
+    public Expr rewriteJoinOnPredicate(Expr predicate) {
```

Comment:
It looks like a lot of repetitive code, maybe you need mark @todo to refactor their ~

Reply:
Of course, it's not important.